python/lsst/daf/butler/direct_query_driver/_result_page_converter.py

```diff
@@ -326,17 +326,35 @@ class GeneralResultPageConverter(ResultPageConverter):  # numpydoc ignore=PR01
 
     def __init__(self, spec: GeneralResultSpec, ctx: ResultPageConverterContext) -> None:
         self.spec = spec
-
-        result_columns = spec.get_result_columns()
+        # In case `spec.include_dimension_records` is True then in addition to
+        # columns returned by the query we have to add columns from dimension
+        # records that are not returned by the query. These columns belong to
+        # either cached or skypix dimensions.
```

Comment:
My idea was that when a client explicitly asks for dimension records then all their columns for all dimensions need to be present in the returned rows.  Handling it differently between cached and non-cached dimensions will probably complicate client code when that code wants to look at individual fields (as opposed to the DatasetRefs returned from iter_tuples). Potentially we can avoid adding those columns to the GeneralResultPage, and make GeneralResult to add them during iteration, but that would require additional round trips for filling caches in case of remote butler.